src/openms/source/ANALYSIS/ID/SiriusAdapterAlgorithm.cpp

```diff
@@ -44,99 +43,471 @@
+                                   "ymdb",
+                                   "keggmine",
+                                   "ecocycmine",
+                                   "ymdbmine"});
+
+      // -I, --ions-enforced=<ionsEnforced>
+      // The iontype/adduct of the MS/MS data. Example: [M+H]+,
+      // [M-H]-, [M+Cl]-, [M+Na]+, [M]+. You can also provide a
+      // comma separated list of adducts.
+      parameter(
+                 SiriusName("ions-enforced"),
+                 DefaultValue(""),
+                 Description("The iontype/adduct of the MS/MS data. Example: [M+H]+, \n"
+                             "[M-H]-, [M+Cl]-, [M+Na]+, [M]+. You can also provide a \n"
+                             "comma separated list of adducts.")
+               );
+
+      // Parameters / Information covered by the .ms file:
+      // -1, --ms1=<ms1> MS1 spectrum file name
+      // -2, --ms2=<ms2> MS2 spectra file names
+      // -z, mz, precursor, --parentmass=<parentMz> The mass of the parent ion
+    }
+
+    void SiriusAdapterAlgorithm::Fingerid::parameters()
+    {
+      // -c, --candidates=<numberOfCandidates>
+      // Number of molecular structure candidates in the output.
+      parameter(
+                 FingeridName("candidates"),
+                 DefaultValue(10),
+                 Description("Number of molecular structure candidates in the output.")
+               ).withMinInt(1);
+
+      // -d, --db , --fingeriddb, --fingerid-db, --fingerid_db=<database>
+      // Search structure in given database. By default the same database
+      // for molecular formula search is also used for structure search.
+      // If no database is used for molecular formula search, PubChem is
+      // used for structure search.
+
+      parameter(
+                 FingeridName("db"),
+                 DefaultValue("all"),
+                 Description("Search structure in given database.")
+                ).withValidStrings({"all",
+                                    "pubchem",
+                                    "mesh",
+                                    "hmdb",
+                                    "knapsack",
+                                    "chebi",
+                                    "pubmed",
+                                    "bio",
+                                    "kegg",
+                                    "hsdb",
+                                    "maconda",
+                                    "metacyc",
+                                    "gnps",
+                                    "zincbio",
+                                    "train",
+                                    "undp",
+                                    "pantcyc",
+                                    "ymdb",
+                                    "keggmine",
+                                    "ecocycmine",
+                                    "ymdbmine",
+                                    "custom",
```

Comment:
Any help on how to use the custom database?

Reply:
I can add that to the description. As far as I know, they have to be renamed to custom_n and then copied to a specific directory.